anthropic-model-provider-go/main.go

```diff
@@ -0,0 +1,54 @@
+package main
+
+import (
+	"fmt"
+	"net/http"
+	"net/http/httputil"
+	"os"
+
+	aproxy "github.com/obot-platform/tools/anthropic-model-provider/proxy"
+	"github.com/obot-platform/tools/openai-model-provider/proxy"
+)
+
+func main() {
+	isValidate := len(os.Args) > 1 && os.Args[1] == "validate"
+
+	cfg := &proxy.Config{
+		APIKey:          os.Getenv("OBOT_ANTHROPIC_MODEL_PROVIDER_API_KEY"), // optional, as e.g. Ollama doesn't require an API key
+		ListenPort:      os.Getenv("PORT"),
+		BaseURL:         "https://api.anthropic.com/v1/",
+		RewriteModelsFn: aproxy.RewriteModelsResponse,
+		RewriteHeaderFn: func(header http.Header) {
+			header.Del("Authorization")
+			header.Set("x-api-key", os.Getenv("OBOT_ANTHROPIC_MODEL_PROVIDER_API_KEY"))
+			header.Set("anthropic-version", "2023-06-01")
+		},
+		Name: "Anthropic",
+	}
+
+	prox := aproxy.NewServer(cfg)
+	reverseProxy := &httputil.ReverseProxy{
+		Director: prox.AnthropicProxyRedirect,
+	}
+	reverseProxyModels := &httputil.ReverseProxy{
+		Director:       prox.AnthropicProxyRedirect,
+		ModifyResponse: aproxy.RewriteModelsResponse,
+	}
+	cfg.CustomPathHandleFuncs = map[string]http.HandlerFunc{
+		"/v1/models": reverseProxyModels.ServeHTTP,
+		"/v1/":       reverseProxy.ServeHTTP,
+	}
```

Comment:
There seem to be a few ways to set the model response re-writer now. After this goes in, and we have some time, someone else should look at that.